src/isomorphic/ReactDebugTool.js

```diff
@@ -58,6 +59,10 @@ var ReactDebugTool = {
   onSetState() {
     emitEvent('onSetState');
   },
+  onInstantiateComponent(internalInstance) {
+    ReactDebugInstanceMap.registerInstance(internalInstance);
```

Comment:
I want to make sure this happens before any devtool receives the event. (Since devtools are actually going to get IDs—I will implement this separately later but #6046 shows how it could work).

I also want to have the opposite guarantee for unmounting: that ID gets unregistered *after* all the other devtools have run. This is why putting such a devtool as first in list wouldn’t work.